logstash-core/src/main/java/org/logstash/common/BufferedTokenizerExt.java

```diff
@@ -106,12 +106,13 @@ public RubyArray extract(final ThreadContext context, IRubyObject data) {
     public IRubyObject flush(final ThreadContext context) {
         final IRubyObject buffer = input.join(context);
         input.clear();
+        inputSize = 0;
```

Comment:
👍🏼 this change _does_ actually fix the issue where `inputSize` is not reset on a flush.